backend/src/test/java/ca/bc/gov/backendstartapi/service/FavouriteActivityServiceTest.java

```diff
@@ -0,0 +1,200 @@
+        activityExists.getMessage());
+  }
+
+  @Test
+  @DisplayName("getAllUserFavoriteActivitiesTest")
+  void getAllUserFavoriteActivitiesTest() {
+    when(loggedUserService.getLoggedUserId()).thenReturn(USER_ID);
+
+    FavouriteActivityEntity activityOne = new FavouriteActivityEntity();
+    FavouriteActivityEntity activityTwo = new FavouriteActivityEntity();
+    List<FavouriteActivityEntity> activityEntities =
+        new ArrayList<>(List.of(activityOne, activityTwo));
+    when(favouriteActivityRepository.findAllByEnabledAndUserId(true, USER_ID))
+        .thenReturn(activityEntities);
+
+    List<FavouriteActivityEntity> entityList =
+        favouriteActivityService.getAllUserFavoriteActivities();
+
+    Assertions.assertFalse(entityList.isEmpty());
+  }
+
+  @Test
+  @DisplayName("updateUserActivityTest")
+  void updateUserActivityTest() {
+    when(loggedUserService.getLoggedUserId()).thenReturn(USER_ID);
+
+    FavouriteActivityEntity entity = new FavouriteActivityEntity();
+    entity.setActivity(ActivityEnum.SEEDLING_REQUEST);
+    entity.setHighlighted(false);
+    entity.setEnabled(true);
+    when(favouriteActivityRepository.findById(any())).thenReturn(Optional.of(entity));
+
+    when(favouriteActivityRepository.save(any())).thenReturn(entity);
+
+    FavouriteActivityUpdateDto updateDto = new FavouriteActivityUpdateDto(true, true);
+    FavouriteActivityEntity saved = favouriteActivityService.updateUserActivity(1L, updateDto);
+
+    Assertions.assertNotNull(saved);
+  }
+
+  @Test
+  @DisplayName("updateUserActivityExceptionTest")
+  void updateUserActivityExceptionTest() {
+    when(loggedUserService.getLoggedUserId()).thenReturn(USER_ID);
+
+    FavouriteActivityEntity entity = new FavouriteActivityEntity();
+    entity.setActivity(ActivityEnum.SEEDLING_REQUEST);
+    entity.setHighlighted(false);
+    entity.setEnabled(true);
+    when(favouriteActivityRepository.findById(any())).thenReturn(Optional.empty());
+
+    when(favouriteActivityRepository.save(any())).thenReturn(entity);
+
+    FavouriteActivityUpdateDto updateDto = new FavouriteActivityUpdateDto(true, true);
+
+    Exception e =
+        Assertions.assertThrows(
+            ActivityNotFoundException.class,
+            () -> favouriteActivityService.updateUserActivity(1L, updateDto));
+
+    Assertions.assertEquals("404 NOT_FOUND \"Activity don't exist!\"", e.getMessage());
+  }
+
+  @Test
+  @DisplayName("deleteUserActivityTest")
+  void deleteUserActivityTest() {
+    when(loggedUserService.getLoggedUserId()).thenReturn(USER_ID);
+
+    FavouriteActivityEntity entity = new FavouriteActivityEntity();
+    entity.setActivity(ActivityEnum.SEEDLING_REQUEST);
+    entity.setHighlighted(false);
+    entity.setEnabled(true);
+    when(favouriteActivityRepository.findById(any())).thenReturn(Optional.of(entity));
+
+    doNothing().when(favouriteActivityRepository).deleteById(any());
+
+    favouriteActivityService.deleteUserActivity(1L);
+  }
```

Comment:
No assertion made here either, though I don't see what we could check since we're mocking the repository. I think we can keep just the `deleteUserActivityExceptionTest` for the delete case for now: we'd only lose one line of coverage, which is the call to the mocked `FavouriteActivityRepository.deleteById`.

Reply:
Yup. I added an assertion so we can keep that line :)